src/lando/main/scm/hg_scm.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from functools import wraps
+from io import StringIO
+from pathlib import Path
+from typing import ContextManager, Optional
+
+import hglib
+
+from lando.api.legacy.hg import REJECTS_PATH, HgRepo
+
+from . import AbstractScm, ScmException
+
+logger = logging.getLogger(__name__)
+
+
+def remap_exceptions(f):
+    """Decorator to remap HgException to ScmException."""
+
+    @wraps(f)
+    def wrapped(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except Exception as e:
+            raise ScmException.from_legacy_exception(e)
+
+    return wrapped
+
+
+class HgScm(AbstractScm):
+
+    path: str
+    hg: HgRepo
+
+    def __init__(self, path: str):
+        self.path = path
+        self.hg = HgRepo(path)
```

Comment:
How is the `HgRepo` different from the `HgScm`? It looks like all the custom methods on this class can just be moved there, and `HgRepo` can subclass `AbstractScm`.

Reply:
The HgRepo is the legacy implementation. The HgScm should ultimately absorb all the logic of the HgRepo, which will allow us to get rid of the legacy bit.